### PR TITLE
Firefly | Bring your cloud Up-to-Code

### PR DESCRIPTION
**Pull request by sefi-infralight**
### What does this PR do?

Add a new integration for [firefly](https://www.gofirefly.io/)
### Motivation

Firefly is a cloud assets management solution that brings **everything** Up-to-Code. Firefly enables DataDog users to fully codify dashboards, users, monitors, etc., to Infrastructure-as-Code (such as Terraform).

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: firefly/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Firefly
+
+## Overview
+Firefly is a Cloud Asset Management solution that helps Cloud teams to discover all their cloud footprint (incl. AWS, GCP, Kubernetes, and Datadog), automatically transform resources to Infrastructure-as-Code, detect drifts, and policy violations to make sure their cloud is aligned with its desired state.
```
Comment (anawishnoff) [inline firefly/README.md:4]:
```suggestion
Firefly is a Cloud Asset Management solution that helps Cloud teams discover their cloud footprint (incl. AWS, GCP, Kubernetes, and Datadog), automatically transform resources to Infrastructure-as-Code, and detect drifts and policy violations to make sure their cloud is aligned with its desired state.
```

File: firefly/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Firefly helps teams make their Datadog assets immutable, versioned, scalable and monitored by managing them as code in any Infrastructure-as-Code (IaC) tool of their choice.
+
+### Full Cloud Inventory
+Get a fully searchable inventory of all Datadog assets, alongside other cloud assets, such as AWS, K8s, GCP, Okta, and more.
+
+### Automatic Codification
+Automatically turn a single or multiple Datadog assets into code, including  Terraform, Pulumi, Cloudformation, and CDK.
```
Comment (anawishnoff) [inline firefly/README.md:11]:
```suggestion
Automatically turn Datadog assets into code, including Terraform, Pulumi, Cloudformation, and CDK.
```

Comment (anawishnoff) [reply firefly/README.md:11]: Are Terraform, Pulumi, and Cloudformation examples of IaC tools that you can manage these assets in? I think some clarity on these examples would be helpful

File: firefly/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### Detect and Fix Drifts
+Get real-time notifications on any discrepancy and fix it as it occurs, directly to your repository.
```
Comment (anawishnoff) [inline firefly/README.md:14]:
```suggestion
Get real-time notifications on any discrepancy and push a fix directly to your repository as it occurs.
```

File: firefly/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### Detect and Fix Policy Violations
+Get alerts and fix any policy violation, using pre-made or custom policies, using Firefly unified policy engine, to find hazardous misconfigurations or costly underutilizations.
```
Comment (anawishnoff) [inline firefly/README.md:17]:
```suggestion
Use the Firefly unified policy engine to find hazardous misconfigurations or costly underutilizations, and get alerts on policy violations for both custom and pre-made policies.
```

File: firefly/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+## Setup
+
+### Configure Firefly - Datadog Integration:
+1. From your Firefly UI, go to: **Settings > Integrations > Datadog**
+2. In order to create the integration you will have to:
+    * Create a new application key
+      * Copy the application key and paste it into the dedicated line
+    * Create a new API key
+      * Copy the application key and paste it into the dedicated line
+3. Click done
```
Comment (anawishnoff) [inline firefly/README.md:28]:
```suggestion
3. Click **Done**
```

File: firefly/manifest.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  "tile": {
+    "overview": "README.md#Overview",
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Bring your cloud Up-to-Code",
+    "title": "Firefly",
+    "media": [
+      {
+        "caption": "Full Cloud Inventory",
```
Comment (anawishnoff) [inline firefly/manifest.json:15]: The images look a bit small and are pixelated when zooming in - would you be able to increase the resolution on these?

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few initial comments, thanks!

File: .azure-pipelines/all.yml [hunk 7]
```diff
@@ -132,10 +132,10 @@ jobs:
       displayName: Puma
       os: linux
     - checkName: purefa
-      displayName: purefa
```
Comment (sarah-witt) [inline .azure-pipelines/all.yml:135]: If you update your branch with master, and remove these changes, validation should pass.

**Comment by anawishnoff** [comment]
Hi @sefi-infralight, it looks like the images are quite blurry. Would you be able to re-upload higher resolution images to this PR? Thanks!

File: firefly/README.md (hunk 2 above)
Comment (sefi-infralight) [reply firefly/README.md:11]: Firefly's ability is to describe DataDog assets as a code (Terraform / pulumi / cdk, etc) and manage them on Firefly.
The management itself of the resources is via Firefly and not solely through these tools as it applies in your phrasing

[the remark this answers is not in the document]
Comment (sefi-infralight) [reply firefly/README.md:11]: Firefly's ability is to describe DataDog assets as a code (Terraform / pulumi / cdk, etc) and manage them on Firefly.
The management itself of the resources is via Firefly and not solely through these tools as it applies in your phrasing

File: firefly/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - Firefly
+
+## 1.0.0
+
+### Changes
+
+* [FEATURE] Initial release - Firefly Datadog Application
```
Comment (anawishnoff) [inline firefly/CHANGELOG.md:7]:
```suggestion
* [FEATURE] Initial release - Firefly Datadog Integration
```

File: firefly/README.md (hunk 2 above)
Comment (anawishnoff) [reply firefly/README.md:11]: No problem, we can keep this as-is.

File: firefly/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+Firefly helps teams make their Datadog assets immutable, versioned, scalable, and monitored by managing them as code in any Infrastructure-as-Code (IaC) tool of their choice.
+
+### Full cloud inventory
+Get a fully searchable inventory of all Datadog assets alongside other cloud assets, such as AWS, K8s, GCP, Okta, and more.
+
+### Automatic codification
+Automatically turn a single or multiple Datadog assets into code, including  Terraform, Pulumi, Cloudformation, and CDK.
+
+### Detect and fix drift
+Get real-time notifications on any discrepancy and push a fix directly to your repository as it occurs.
```
Comment (anawishnoff) [inline firefly/README.md:14]: Could you specify what kind of discrepancy would send a notification?

File: firefly/README.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+### Detect and Fix Policy Violations
+Use the Firefly unified policy engine to find hazardous misconfigurations or costly underutilizations, and get alerts on policy violations for both custom and pre-made policies.
+
+## Setup
+
+### Configure Firefly - Datadog integration
+1. Create a new Datadog application key and API key.
+2. In the Firefly UI, go to: **Settings > Integrations > Datadog**
+3. Copy the application key and paste it into the dedicated line.
+4. Copy the application key and paste it into the dedicated line.
```
Comment (anawishnoff) [inline firefly/README.md:25]:
```suggestion
4. Copy the API key and paste it into the dedicated line.
```

File: firefly/manifest.json [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+      }
+    ],
+    "classifier_tags": [
+      "Category::Autodiscovery",
+      "Category::Automation",
+      "Category::Cloud",
+      "Category::Developer Tools",
+      "Category::Google Cloud",
+      "Category::AWS",
+      "Offering::Integration"
```
Comment (anawishnoff) [inline firefly/manifest.json:42]: You may also want to add supported OSes here, such as:

```suggestion
      "Offering::Integration",
      "Supported OS::Linux",
      "Supported OS::Mac OS",
      "Supported OS::Windows"
```

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Just a few more comments from my end

File: firefly/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (urseberry) [reply firefly/README.md:11]: Ah, I see. In that case, how about this:
```suggestion
Automatically turn single or multiple Datadog assets into code, including Terraform, Pulumi, Cloudformation, and CDK specifications.
```

File: firefly/README.md (hunk 9 above)
Comment (sefi-infralight) [reply firefly/README.md:14]: The discrepancy is between the state of the cloud and what’s described in the IaC code, usually it’s referred to as “Drift”

File: firefly/README.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
+# Firefly
+
+## Overview
+Firefly is a Cloud Asset Management solution that helps Cloud teams discover their cloud footprint (incl. AWS, GCP, Kubernetes, and Datadog), automatically transform resources to Infrastructure-as-Code, and detect drifts and policy violations to make sure their cloud is aligned with its desired state.
```
Comment (urseberry) [inline firefly/README.md:4]:
```suggestion
Firefly is a Cloud Asset Management solution that helps Cloud teams discover their cloud footprint (including AWS, GCP, Kubernetes, and Datadog), automatically transform resources to Infrastructure-as-Code, and detect drifts and policy violations to make sure their cloud is aligned with its desired state.
```

File: firefly/README.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+Firefly helps teams make their Datadog assets immutable, versioned, scalable, and monitored by managing them as code in any Infrastructure-as-Code (IaC) tool of their choice.
+
+### Full cloud inventory
+Get a fully searchable inventory of all Datadog assets alongside other cloud assets, such as AWS, K8s, GCP, Okta, and more.
+
+### Automatic codification
+Automatically turn single or multiple Datadog assets into code, including Terraform, Pulumi, Cloudformation, and CDK specifications.
+
+### Detect and fix drift
+Get real-time notifications on any discrepancy and push a fix directly to your repository as it occurs.
```
Comment (urseberry) [inline firefly/README.md:14]:
```suggestion
Get real-time notification of any discrepancy between the cloud and your specification, and push a fix directly to your repository as drift occurs.
```

File: firefly/README.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+### Configure Firefly - Datadog integration
+1. Create a new Datadog application key and API key.
+2. In the Firefly UI, go to: **Settings > Integrations > Datadog**
+3. Copy the application key and paste it into the dedicated line.
+4. Copy the API key and paste it into the dedicated line.
+5. Click **Done**.
+
+## Support
+Any questions? We are here to help! Email [contact@gofirefly.io][1] or use the in-app chat.
```
Comment (urseberry) [inline firefly/README.md:29]:
```suggestion
Any questions? Email [contact@gofirefly.io][1] or use the in-app chat.
```

**Review by urseberry — COMMENTED** [review]
Please also update the images. Several are blurry.

When you retake the screenshots, please keep in mind that the Datadog documentation page usable space is narrow. If possible, zoom in on the UI to 150%. Take a screenshot of only the active pane, and leave out sidebars and background pages.

File: firefly/README.md (hunk 9 above)
Comment (anawishnoff) [reply firefly/README.md:14]: Gotcha, it would be great if you could add that context directly into the description here. Thanks!

Comment (sefi-infralight) [reply firefly/README.md]: Get real-time notification of any discrepancy between your Infrastructure-as-Code and your actual state of cloud, and push a fix directly to your repository as drift occurs.

**Review by urseberry — APPROVED** [review]
Docs team approves the copy. The images are clear now, though they are still very wide.

**Review by anawishnoff — COMMENTED** [review]
This looks good, I left one small nit on the copy. Otherwise, note that your images will likely get cut off in the preview on the tile. Images must have an aspect ratio of 16:9 in order to not get cut off in the gallery preview on the tile - feel free to add your images onto a background to combat this.

**Review by anawishnoff — APPROVED** [review]

**Comment by sefi-infralight** [comment]
Hi, can you please merge our PR? @anawishnoff @urseberry

**Pull request merged**